### PR TITLE
feat: revocation by document hash

### PR DESCRIPTION
**Pull request by HJunyuan**
## Context
- This reference implementation started off with revocation by document ID
- In order to align with how we perform revocation of documents issued onto the Ethereum document store, revocation by merkle root and intermediate hashes should be performed by the OCSP Responder (i.e. do not revoke by document id)

## What does this PR do?

- [x] Refactor naming convention from ~~`documentId`~~ -> `documentHash`
- [x] Modify API response body: `revoked = true / false`, `success = true / false`, etc.
- [x] Add plugins to support local development: `npm run dev`
- [x] Improve readability of README
- [x] Serverless deployment example: `npm run deploy` (i.e. `sls deploy`)

### CODE REVIEW

File: src/functions/insert/handler.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import createError from "http-errors";
+
+import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { formatJSONResponse } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { client } from "@services/dynamoDb";
+
+import schema from "./schema";
+
+const REASON_CODES = [...Array(11).keys()];
```
Comment (cavacado) [inline src/functions/insert/handler.ts]: clever, but reads poorly, could you wrap it around a generate range function and put it in utils?

**Review by cavacado — APPROVED** [review]

File: src/functions/insert/handler.ts (hunk 1 above)
Comment (HJunyuan) [reply src/functions/insert/handler.ts]: Certainly, will change it to:

```suggestion
const REASON_CODES = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
```

**Review by cavacado — APPROVED** [review]
👍 lgtm lets go

**Comment by Abdulhakimsg** [comment]
Thanks for this feature! Extremely useful for our team

- In order to align with how we perform revocation of documents issued onto the Ethereum document store, revocation by merkle root and intermediate hashes should be performed by the OCSP Responder (i.e. do not revoke by document id)


Alt Text

**Pull request merged**